invoices/utils.py

```diff
@@ -277,3 +318,41 @@ def refresh_stats(start_date, end_date):
             invoice.bill_rate_avg = 0
         invoice.save()
         logger.debug("Updated statistics for %s", invoice)
+
+
+def refresh_weekly_stats(start_date, end_date):
+    if start_date and end_date:
+        weekly_reports = WeeklyReport.objects.filter(
+            year__gte=start_date.year,
+            year__lte=end_date.year,
+            week__gte=start_date.isocalendar()[1],
+            week__lte=end_date.isocalendar()[1])
+        logger.info("Updating statistics for weekly reports between %s and %s: %s invoices",
+                    start_date,
+                    end_date,
+                    weekly_reports.count())
+    else:
+        logger.info("Updating statistics for all weekly reports")
+        weekly_reports = WeeklyReport.objects.all()
+    for weekly_report in weekly_reports:
+        entries = HourEntry.objects.filter(weekly_report=weekly_report).filter(incurred_hours__gt=0)
+        aws_entries = None
+        stats = calculate_weekly_entry_stats(entries, aws_entries)
+        for field in STATS_FIELDS:
```

Comment:
The rest of the method seems to be exact copy of what is in `refresh_stats`, so splitting that to a well-named method would make sense, no?